modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -190,6 +193,10 @@
 import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.CORRUPTED_DATA_FILES_MNTC_TASK_NAME;
 import static org.apache.ignite.internal.util.IgniteUtils.GB;
 import static org.apache.ignite.internal.util.IgniteUtils.checkpointBufferSize;
+import static org.apache.ignite.transactions.TransactionState.COMMITTED;
```

Comment:
I don't like the fact the we add a lot of details about transactions into DatabaseSharedManager which be aware of this stuff as little as possible.

Why not to introduce a filter (e.g. we may call it `NonFinishedTxFilter`) and put it into IgniteTxManager. All the structures like `uncommitedTx` set and logic will go there, database manager will just notify the filter about TX records and ask whether next transaction should be skipped.

What do you think about that?

Reply:
huge +1 ! thanks for pointing this !